s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -263,4 +338,21 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
           Future.failed(new S3Exception(err))
         }
     }
+
+  private def sizeFromHeaders(headers: Seq[HttpHeader]): Option[Long] =
+    headers.find(_.lowercaseName() == CONTENT_LENGTH.toLowerCase).map(_.value().toLong)
+
+  private def etagFromHeaders(headers: Seq[HttpHeader]): Option[String] =
+    headers.find(_.lowercaseName() == ETAG.toLowerCase).map(_.value())
+
+  private def lastModifiedFromHeaders(headers: Seq[HttpHeader]): Option[Instant] =
+    headers
+      .find(_.lowercaseName() == LAST_MODIFIED.toLowerCase)
+      // FIXME we have problems reading the time format (Thu, 02 Nov 2017 23:47:39 GMT)
+      .flatMap(h => Try(Instant.parse(h.value())).toOption)
+
+  private def storageClassFromHeaders(headers: Seq[HttpHeader]): Option[StorageClass] =
+    headers.find(_.lowercaseName() == STORAGE_CLASS.toLowerCase).flatMap { header =>
+      StorageClass(header.value)
+    }
```

Comment:
I can't see that these are used anywhere, and I think that if they are they should use the existing header APIs of the Akka HTTP model instead. (somethingl like ``headers.collectFirst { case `Last-Modified(dateTime)` => dateTime }`` for example)